app/controllers/admin/cases_controller.rb

```diff
@@ -97,7 +87,43 @@ def update
         @case.publisher.status_updates.create(status: PublisherStatusUpdate::ACTIVE, publisher_note: note)
       end
 
-      redirect_to admin_cases_path
+      return redirect_to [:admin, @case], flash: { notice: "Case has been moved back to in progress"} if params[:status] == Case::IN_PROGRESS
+
+      next_case = Case.where(assignee: current_user, status: Case::IN_PROGRESS).take
+      if next_case
+        redirect_to [:admin, next_case], flash: { notice: "Previous case #{@case.number} has been marked #{params[:status]}"}
+      else
+        redirect_to admin_cases_path
+      end
+    end
+
+    private
+
+    def parse_search(search_case, queries)
+      statuses = queries.select { |x| x.include?("status") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+      assigned = queries.select { |x| x.include?("assigned") }.map { |x| x.split(':').last&.gsub(/[^0-9a-z_]/i, '') }
+
+      first_status = statuses[0]
+      if first_status.present?
+        value = first_status
+        search_case = search_case.where('status LIKE ?', "%#{value}%")
```

Comment:
Instead of `status LIKE %value%`, would it be possible to use `where(status: value)` or `status LIKE value%`? Otherwise it'll be an O(n) search operation. 

From https://www.postgresql.org/docs/9.3/indexes-types.html
```
The optimizer can also use a B-tree index for queries involving the pattern matching operators LIKE and ~ if the pattern is a constant and is anchored to the beginning of the string — for example, col LIKE 'foo%' or col ~ '^foo', but not col LIKE '%bar'. 
```

Reply:
Yep! https://github.com/brave-intl/publishers/pull/2627/commits/0bb03e74fb8d1cbfbb6fb6ac5e649d264defde29